legend-engine-core/legend-engine-core-language-pure/legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/extension/ConnectionFactoryExtension.java

```diff
@@ -51,7 +51,7 @@ default Optional<Connection> tryBuildFromConnection(Connection connection, Strin
         return Optional.empty();
     }
 
-    default Optional<Pair<Connection, List<Closeable>>> tryBuildTestConnection(Connection sourceConnection, EmbeddedData data)
+    default Optional<Pair<Connection, List<Closeable>>> tryBuildTestConnection(Connection sourceConnection, List<EmbeddedData> data)
```

Comment:
Let's not break the existing contract. Let's add another method maybe somethign like `tryBuildTestConnectionWithMultiData`
If the size of the list is 1, we should still default to `tryBuildTestConnection` as the defaullt implementation. 
This would prevent us from doing ->get(0) on the implementations

Reply:
I think we need to break the contract because we do an extension collect for that function and it expects a list. I can do a size 1 check to prevent any funkines with the get(0) or make it iterative